data_models/data_refinery_models/models/organism.py

```diff
@@ -0,0 +1,121 @@
+import requests
+from xml.etree import ElementTree
+from django.db import models
+from data_refinery_models.models.base_models import TimeTrackedModel
+
+# Import and set logger
+import logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+NCBI_ROOT_URL = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/"
+ESEARCH_URL = NCBI_ROOT_URL + "esearch.fcgi"
+EFETCH_URL = NCBI_ROOT_URL + "efetch.fcgi"
+
+
+class UnscientifcNameError(BaseException):
+    pass
+
+
+class InvalidNCBITaxonomyId(BaseException):
+    pass
+
+
+def get_scientific_name(taxonomy_id: int):
+    parameters = {"db": "taxonomy", "id": str(taxonomy_id)}
+    response = requests.get(EFETCH_URL, parameters)
+
+    root = ElementTree.fromstring(response.text)
+    taxon_list = root.findall("Taxon")
+
+    if(len(taxon_list) == 0):
+        logger.error("No names returned by ncbi.nlm.nih.gov for organism "
+                     + "with taxonomy ID %d.",
+                     taxonomy_id)
+        raise InvalidNCBITaxonomyId
+
+    return taxon_list[0].find("ScientificName").text.upper()
```

Comment:
I'm going to say yes. Of course it's always possible for the API to have a bug and fail here, but it's an old API which is widely used so I'm going to go ahead and trust them.